src/persistentReducedStream.ts

```diff
@@ -14,26 +13,68 @@ import { defaultErrorSubject } from './internal/defaultErrorSubject';
 import { markName } from './internal/markers';
 import { tag } from 'rxjs-spy/operators';
 
-/**
- * A hot, persistent observable with a state field
- */
-export class StateStream<State> extends Observable<State>
-  implements SubscriptionLike {
+export class PersistentReducedStateStream<State> extends Observable<State> {
   private subject: BehaviorSubject<State>;
-  private subscription: Subscription;
-
-  constructor(initialState: State, reducedState$: Observable<State>) {
-    super();
-    this.subject = new BehaviorSubject(initialState);
+  private reducerSubscription?: Subscription;
+  public name: string;
+  private reducers: RegisteredReducer<State, any>[];
+  private errorSubject: Subject<any>;
 
+  /**
+   * Subscribes to the action stream and starts reducing this state stream.
+   * startReducing does not always immediately emit a value since the value
+   * might have already been emitted when the state stream was created.
+   *
+   * initialState can be passed to reset the state of the stream to some other value.
+   * This can be useful for hot-reloading
+   *
+   * startReducing will recreate the inner subject if it was previously closed
+   */
+  startReducing = (action$: ActionStream, initialState?: State) => {
+    // Reset the subject value if it doesn't match the desired initialState
+    // This is normally used for hot-reloading
+    if (this.subject.closed && initialState) {
+      this.subject = new BehaviorSubject(initialState);
+    } else if (initialState && this.subject.value !== initialState) {
+      this.subject.next(initialState);
+    }
     // NOTE: .subscribe(this.subject) does not work
     // All tests pass with it, but in practice there are cases
     // where the subject isn't updated as expected
-    this.subscription = reducedState$.subscribe({
-      complete: () => this.subject.complete(),
-      error: (err) => this.subject.error(err),
-      next: (state) => this.subject.next(state),
-    });
+    this.reducerSubscription = action$
+      .pipe(
+        combineReducers(initialState || this.subject.value, this.reducers, {
+          errorSubject: this.errorSubject,
+        }),
+        markName(this.name),
+        tag(this.name)
+      )
+      .subscribe({
+        complete: () => this.subject.complete(),
+        error: (err) => this.subject.error(err),
+        next: (state) => this.subject.next(state),
+      });
+    return this;
+  };
+
+  stopReducing = () => {
+    this.subject.unsubscribe();
```

Comment:
When reducing is stopped, the underlying `subject` will be closed. This means that reading the state of the state stream will throw an error (which is good because this might be outdated state).